sentinel-adapter/sentinel-dubbo-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/dubbo/origin/DubboOriginRegistryTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 1999-2020 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.csp.sentinel.adapter.dubbo.origin;
+
+import com.alibaba.csp.sentinel.adapter.dubbo.DubboUtils;
+import com.alibaba.dubbo.rpc.Invocation;
+import com.alibaba.dubbo.rpc.Invoker;
+import com.alibaba.dubbo.rpc.RpcInvocation;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author tiecheng
+ */
+public class DubboOriginRegistryTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testDefaultOriginParserFail() {
+        DubboOriginParserRegistry.getDubboOriginParser().parse(null, null);
+    }
+
+    @Test
+    public void testDefaultOriginParserSuccess() {
+        RpcInvocation invocation = new RpcInvocation();
+        String dubboName = "sentinel";
+        invocation.setAttachment(DubboUtils.DUBBO_APPLICATION_KEY, dubboName);
+        String origin = DubboOriginParserRegistry.getDubboOriginParser().parse(null, invocation);
+        Assert.assertEquals(dubboName, origin);
+    }
+
+    @Test
+    public void testCustomOriginParser() {
+        DubboOriginParserRegistry.setDubboOriginParser(new DubboOriginParser() {
+            @Override
+            public String parse(Invoker<?> invoker, Invocation invocation) {
+                return invocation.getAttachment(DubboUtils.DUBBO_APPLICATION_KEY, "default") + "_" + invocation
+                    .getMethodName();
+            }
+        });
+
+        RpcInvocation invocation = new RpcInvocation();
+        String origin = DubboOriginParserRegistry.getDubboOriginParser().parse(null, invocation);
+        Assert.assertEquals("default_null", origin);
+
+        String dubboName = "sentinel";
+        invocation.setAttachment(DubboUtils.DUBBO_APPLICATION_KEY, dubboName);
+        origin = DubboOriginParserRegistry.getDubboOriginParser().parse(null, invocation);
+        Assert.assertEquals(dubboName + "_null", origin);
+
+        invocation.setMethodName("hello");
+        origin = DubboOriginParserRegistry.getDubboOriginParser().parse(null, invocation);
+        Assert.assertEquals(dubboName + "_hello", origin);
+
+        // if not set default, effect on SentinelDubboProviderFilterTest when mvn integration-test
+        DubboOriginParserRegistry.setDubboOriginParser(new DefaultDubboOriginParser());
+    }
```

Comment:
It may better that we use uniform clean method before and after each test case.
such as:

```java
@Before
public void setUp() {
    DubboOriginParserRegistry.setDubboOriginParser(new DefaultDubboOriginParser());
}

@After
public void cleanUp() {
    DubboOriginParserRegistry.setDubboOriginParser(new DefaultDubboOriginParser());
}
```